astropy/visualization/interval.py

```diff
@@ -226,9 +227,10 @@ class ZScaleInterval(BaseInterval):
         5.
     """
 
-    def __init__(self, nsamples=1000, contrast=0.25, max_reject=0.5,
-                 min_npixels=5, krej=2.5, max_iterations=5):
-        self.nsamples = nsamples
+    def __init__(self, n_samples=1000, contrast=0.25, max_reject=0.5,
+                 min_npixels=5, krej=2.5, max_iterations=5, nsamples=None):
```

Comment:
Thanks! I think you should look into using this:

https://github.com/astropy/astropy/blob/d4fc9d9b96f0879c363a8183842e5cf6067fae37/astropy/utils/decorators.py#L278

Also subpackage maintainer should contemplate this API change.